### ISSUE TITLE
Cumulative Layout Shift

### ISSUE THREAD

**Issue by bdekoz** [mozilla/standards-positions#386]
## Request for Mozilla Position on an Emerging Web Specification

* Specification Title: Cumulative Layout Shift
* Specification or proposal URL: https://web.dev/cls/
* Caniuse.com URL (optional): 
* Bugzilla URL (optional): 
* Mozillians who can provide input (optional): Daniel Holbert, Cameron McCormack

### Other information

One of 3 metrics defined by Google's Web Vitals in  2020. See:  https://web.dev/vitals/

This depends on the layout instability API, for which there is a separate issue [#374](https://github.com/mozilla/standards-positions/issues/374). 

More detailed specs of 'layout-shift-score' and links to canonical sources will be added as comments to this issue.

**Comment by npm1** [comment]
To clarify, the specification would be https://wicg.github.io/layout-instability/ since CLS is not exposed directly but can be computed based on the layout shifts exposed by the Layout Instability API. We can add a section dedicated to CLS if it helps.

**Comment by dholbert** [comment]
Duplicate of https://github.com/mozilla/standards-positions/issues/374 ?

**Comment by dholbert** [comment]
(Oh, now I see `This depends on the layout instability API, for which there is a separate issue #374`. So I guess not a dupe, but related.)

**Comment by bdekoz** [comment]
@npm1 yes, it would be helpful to have a section to CLS and how it's computed in the proposal, just because that will clarify and guide future work.

**Comment by dholbert** [comment]
One concern: it's hard to harmonize these two statements:

> Developers are **advised to avoid relying on the precise value** of the layout shift score, as the metric might evolve over time, and user agents might compromise precision in the interest of calculation efficiency. (source: [WICG spec text](https://wicg.github.io/layout-instability/#example))

...vs:
>  To provide a good user experience, pages should **maintain a CLS of less than 0.1**. (source: [Web.dev article](https://web.dev/cls/), [another web.dev article](https://web.dev/vitals/), [page speed insights for e.g. mozilla.org](https://developers.google.com/speed/pagespeed/insights/?url=mozilla.org)

(The spec text is about individual LS values rather than the CLS, but it applies to the CLS as well by extension, since the CLS is a sum of LS scores.)

I read the above-quoted spec text as saying that the precise value (and precise thresholds like 0.1) aren't meaningful in isolation, and that authors should rather watch for increases/decreases rather than looking at the precise value.  This obviously disagrees with the web.dev articles and the pagespeed report, though, which treats 0.1 as a special threshold.  So: if my reading of the spec isn't the intended meaning, then maybe that line of spec text should be rewritten to clarify what it's trying to say?

**Comment by skobes** [comment]
In https://github.com/WICG/layout-instability/pull/58, the spec for the Layout Instability API has been updated with further details in non-normative text about the definition of CLS.
